plugins/woocommerce-admin/client/products/fields/attribute-field/attribute-field.tsx

```diff
@@ -82,29 +94,51 @@ export const AttributeField: React.FC< AttributeFieldProps > = ( {
 	);
 
 	useEffect( () => {
-		if ( ! value || hydrationComplete ) {
+		// Temporarily always doing hydration, since otherwise new attributes
+		// get removed from Options and Attributes when the other list is then
+		// modified
+		//
+		// This is because the hydration is out of date -- the logic currently
+		// assumes modifications are only made from within the component
+		//
+		// I think we'll need to move the hydration out of the individual component
+		// instance. To where, I do not yet know... maybe in the form context
+		// somewhere so that a single hydration source can be shared between multiple
+		// instances? Something like a simple key-value store in the form context
+		// would be handy.
```

Comment:
We should pull hydration out of this component and have it done somewhere it can be shared with multiple instances of `AttributeField`.

Reply:
Agreed on this comment.  I'd prefer not to see any direct fetching or updating happening in this component, but rather passing down `attributes`, `onAdd`, `onDelete`, etc. and allowing the consumer to dictate this.

This will make these components simpler, more flexible, and easier to test.